internal/queue/queued_channel.go

```diff
@@ -30,14 +43,56 @@ func NewQueuedChannel[T any](channelBufferSize int, capacity int) *QueuedChannel
 	return queue
 }
 
-func (q *QueuedChannel[T]) Queue(items ...T) bool {
-	return q.queue.PushMany(items...)
+func (q *QueuedChannel[T]) Enqueue(items ...T) bool {
+	if q.closed.Load() == closed {
+		return false
+	}
+
+	q.cond.L.Lock()
+	defer q.cond.L.Unlock()
+
+	q.items = append(q.items, items...)
+
+	q.cond.Broadcast()
+
+	return true
 }
 
 func (q *QueuedChannel[T]) GetChannel() <-chan T {
 	return q.ch
 }
 
 func (q *QueuedChannel[T]) Close() {
-	q.queue.Close()
+	q.closed.Store(closed)
+
+	q.cond.L.Lock()
+	defer q.cond.L.Unlock()
+
+	q.cond.Broadcast()
+
+	for _, item := range q.items {
```

Comment:
Leftover debug logs?

Reply:
oops yeah! was leftover from when the leak detector was complaining

Reply:
Fixed. Also moved queue out of internal (i want to use it in bridge too)